internal/events/aggregator.go

```diff
@@ -249,6 +255,46 @@ func (ag *aggregator) processPins(ctx context.Context, pins []*fftypes.Pin, stat
 	return err
 }
 
+func (ag *aggregator) checkOnchainConsistency(ctx context.Context, msg *fftypes.Message, pin *fftypes.Pin) (valid bool, err error) {
+	l := log.L(ctx)
+
+	verifierRef := &fftypes.VerifierRef{
+		Type:  ag.verifierType,
+		Value: pin.Signer,
+	}
+
+	if msg.Header.Key == "" || msg.Header.Key != pin.Signer {
+		l.Errorf("Invalid message '%s'. Key '%s' does not match the signer of the pin: %s", msg.Header.ID, msg.Header.Key, pin.Signer)
+		return false, nil // This is not retryable. skip this message
+	}
+
+	// Verify that we can resolve the signing key back to the identity that is claimed in the batch.
+	resolvedAuthor, err := ag.identity.FindIdentityForVerifier(ctx, []fftypes.IdentityType{
+		fftypes.IdentityTypeOrg,
+		fftypes.IdentityTypeCustom,
+	}, msg.Header.Namespace, verifierRef)
+	if err != nil {
+		return false, err
+	}
+	if resolvedAuthor == nil {
+		if msg.Header.Type == fftypes.MessageTypeDefinition &&
+			(msg.Header.Tag == fftypes.SystemTagIdentityClaim || msg.Header.Tag == fftypes.DeprecatedSystemTagDefineNode || msg.Header.Tag == fftypes.DeprecatedSystemTagDefineOrganization) {
+			// We defer detailed checking of this identity to the system handler
+			return true, nil
+		} else if msg.Header.Type != fftypes.MessageTypePrivate {
+			// Only private messages, or root org broadcasts can have an unregistered key
+			l.Errorf("Invalid message '%s'. Author '%s' cound not be resolved: %s", msg.Header.ID, msg.Header.Author, err)
```

Comment:
This is the special case that allows an anonymous key to be used to pin private messages (per the table).

Note that the key must **not** be registered to any identity to use this path. If the key is registered, then it must be registered to the `author` identity in the private message.